src/version/version.rs

```diff
@@ -45,41 +49,60 @@ fn get_installed_version() -> Result<Version, failure::Error> {
 }
 
 fn check_wrangler_versions() -> Result<WranglerVersion, failure::Error> {
-    let current = get_installed_version()?;
-    let latest = get_latest_version(&current.to_string())?;
-    Ok(WranglerVersion { current, latest })
-}
-
-fn get_latest_version(installed_version: &str) -> Result<Version, failure::Error> {
     let config_dir = get_wrangler_home_dir()?;
     let version_file = config_dir.join("version.toml");
     let current_time = SystemTime::now();
-    let latest_version = match fs::read_to_string(&version_file) {
-        Ok(contents) => {
-            let last_checked_version = LastCheckedVersion::from_str(&contents)?;
+
+    let mut checked = false;
+    let current = get_installed_version()?;
+
+    let latest = match get_version_disk(&version_file)? {
+        Some(last_checked_version) => {
             let time_since_last_checked =
                 current_time.duration_since(last_checked_version.last_checked)?;
+
             if time_since_last_checked.as_secs() < ONE_DAY {
-                let version = Version::parse(&last_checked_version.latest_version)?;
-                Some(version)
-            } else {
-                None
+                checked = true;
             }
+            Version::parse(&last_checked_version.latest_version)?
         }
-        Err(_) => None,
+        // If version.toml doesn't exist, fetch latest version
+        None => get_latest_version(&current.to_string(), &version_file, current_time)?,
     };
-    match latest_version {
-        Some(latest_version) => Ok(latest_version),
-        None => {
-            let latest_version = get_latest_version_from_api(installed_version)?;
-            let updated_file_contents = toml::to_string(&LastCheckedVersion {
-                latest_version: latest_version.to_string(),
-                last_checked: current_time,
-            })?;
-            fs::write(&version_file, updated_file_contents)?;
-            Ok(latest_version)
+
+    Ok(WranglerVersion {
+        current,
+        latest,
+        checked,
+    })
+}
+
+/// Reads version out of version file, is `None` if file does not exist
+fn get_version_disk(version_file: &PathBuf) -> Result<Option<LastCheckedVersion>, failure::Error> {
+    let local_version = match fs::read_to_string(&version_file) {
+        Ok(contents) => {
+            let last_checked_version = LastCheckedVersion::from_str(&contents)?;
```

Comment:
I think this fn should return `Option<LastCheckedVersion>` rather than `Result<Option<LastCheckedVersion>, failure::Error>`. I know this was the function signature in the first place, but I can't see a good way for a user to recover from this error; if this goes wrong, it likely means there is an issue with the file format  (we were able to read the file, but its contents were unexpected), and it seems like the best course of action is to return `None` so that the file is rewritten.

Thoughts? cc @EverlastingBugstopper

Reply:
yeah that makes sense - generally if something goes wrong here we should probably just overwrite the file.